apps/web/src/hooks/useAsyncFileDownload.ts

```diff
@@ -0,0 +1,80 @@
+import { useRef, useState } from "react";
+
+import { ACCESS_TOKEN } from "@gc-digital-talent/auth";
+
+type UseAsyncFileDownloadArgs = {
+  url: string;
+  fileName: string;
+};
+
+type UseAsyncFileDownloadData = {
+  fetching: boolean;
+  abort: AbortController["abort"];
+};
+
+type UseAsyncFileDownloadReturn = [
+  UseAsyncFileDownloadData,
+  () => Promise<void>,
+];
+
+/**
+ * Reads a blob stream for a file
+ * downloading it on completion
+ * returning a promise to be handled
+ *
+ */
+function useAsyncFileDownload({
+  url,
+  fileName,
+}: UseAsyncFileDownloadArgs): UseAsyncFileDownloadReturn {
+  const [fetching, setFetching] = useState<boolean>(false);
+  const controller = useRef<AbortController>(new AbortController());
+
+  function downloadFile(): Promise<void> {
+    // Abort any current requests
+    if (fetching) {
+      controller.current.abort();
+    }
```

Comment:
You can test this if you throttle your network. It should cancel any download in progress if you happen to trigger duplicates so you don't get a bunch of files downloading.

Maybe we don't care and can let users download as many as they'd like? :laughing: I just ran into a chrome warning about downloading multiple files so tried using this to mitigate it.

Reply:
Avoiding duplicate queries is definitely a good move!